packages/governance/src/multisig/multisig.cairo

```diff
@@ -0,0 +1,440 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.18.0 (governance/multisig/multisig.cairo)
+
+/// # Multisig Component
+#[starknet::component]
+pub mod MultisigComponent {
+    use core::num::traits::Zero;
+    use core::panic_with_felt252;
+    use crate::multisig::interface::{IMultisig, TransactionID, TransactionStatus};
+    use starknet::account::Call;
+    use starknet::storage::{Map, StorageMapReadAccess, StorageMapWriteAccess};
+    use starknet::storage::{StoragePointerReadAccess, StoragePointerWriteAccess};
+    use starknet::syscalls::call_contract_syscall;
+    use starknet::{ContractAddress, SyscallResultTrait};
+    use starknet::{get_caller_address, get_contract_address};
+
+    #[derive(Drop, starknet::Store)]
+    struct StorableCallInfo {
+        to: ContractAddress,
+        selector: felt252,
+        calldata_len: u32
+    }
+
+    #[storage]
+    pub struct Storage {
+        pub Multisig_threshold: u32,
+        pub Multisig_signers_count: u32,
+        pub Multisig_is_signer: Map<ContractAddress, bool>,
+        pub Multisig_signers_by_index: Map<u32, ContractAddress>,
+        pub Multisig_signers_indices: Map<ContractAddress, u32>,
+        pub Multisig_total_txs: TransactionID,
+        pub Multisig_tx_confirmations: Map<TransactionID, u32>,
+        pub Multisig_tx_confirmed_by: Map<(TransactionID, ContractAddress), bool>,
+        pub Multisig_tx_executed: Map<TransactionID, bool>,
+        pub Multisig_tx_calls_len: Map<TransactionID, u32>,
+        pub Multisig_tx_call_info: Map<(TransactionID, u32), StorableCallInfo>,
+        pub Multisig_tx_call_calldata: Map<(TransactionID, u32, u32), felt252>,
```

Comment:
We shouldn't need to store these many things if we use the same approach we use in timelock and governor, that is storing just an id of the proposal, and passing the calls to each operation to get the id by rehashing the proposal. This should make the module more consistent with our other modules, and also simplify the implementation, since there shouldn't be no need for tracking many of these.

Reply:
Refactored to the suggested approach that goes without storing calls in the contract's storage. Tx ID is computed by hashing calls + salt and calls are expected to be also provided at the moment of tx execution